pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java

```diff
@@ -103,6 +128,38 @@
      */
     List<String> getList(String namespace, TopicDomain topicDomain) throws PulsarAdminException;
 
+    /**
+     * Get the list of topics under a namespace.
+     * <p/>
+     * Response example:
+     *
+     * <pre>
+     * <code>["topic://my-tenant/my-namespace/topic-1",
+     *  "topic://my-tenant/my-namespace/topic-2"]</code>
+     * </pre>
+     *
+     * @param namespace
+     *            Namespace name
+     *
+     * @param bundle
+     *            Namespace bundle
+     *
+     * @param topicDomain
+     *            use {@link TopicDomain#persistent} to get persistent topics
+     *            use {@link TopicDomain#non_persistent} to get non-persistent topics
+     *            Use null to get both persistent and non-persistent topics
+     *
+     * @return a list of topics
+     *
+     * @throws NotAuthorizedException
+     *             Don't have admin permission
+     * @throws NotFoundException
+     *             Namespace does not exist
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    List<String> getList(String namespace, String bundle, TopicDomain topicDomain) throws PulsarAdminException;
```

Comment:
Suggests  to provide the corresponding `getListAsync` method like the others?

Reply:
How about introducing a GetListOption? we might want to add more params in the future, this will make the API looks very bloated

Reply:
added `getListAsync` with param map for future extension.